Sources/StreamChatUI/ChatChannel/ChatChannelVC.swift

```diff
@@ -71,12 +71,23 @@ open class ChatChannelVC: _ViewController,
         messageListVC.delegate = self
         messageListVC.dataSource = self
         messageListVC.client = client
-
-        messageComposerVC.channelController = client.channelController(for: channelController.cid!)
+        
         messageComposerVC.userSearchController = userSuggestionSearchController
+        
+        func setChannelControllerToComposerIfNeeded(cid: ChannelId?) {
+            guard messageComposerVC.channelController == nil else { return }
+            let composerChannelController = channelController.cid.map { client.channelController(for: $0) }
+            messageComposerVC.channelController = composerChannelController
+        }
```

Comment:
I'd make it `private` in `ChatChannelVC` scope to keep the lifecycle methods short